src/pcre2_compile.c

```diff
@@ -1392,6 +1392,7 @@ while (ptr < ptrend && IS_DIGIT(*ptr))
   if (n > max_value)
     {
     *errorcodeptr = max_error;
+    while (ptr < ptrend && IS_DIGIT(*ptr)) ptr++;
```

Comment:
I approve. I think we should do this change, but perhaps in a separate PR.

Reply:
Here, we're advancing `ptr`. We could equally well rewind the `ptr` back to the beginning of the digits, which might actually be better.

But really, it's an almost irrelevant nitpick.

Reply:
Perl does report the bad offset at the end of the number (hence the preference to do the same).

Agree that in this specific case, where the error is recoverable and we are planning to rewind and read the number again, it might make sense to do the opposite though, but it might be a surprise to future users that expect `ptr` to point past the error from other similar functions.

Reply:
Great! Let's advance the pointer then, for consistency.